src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -18,600 +30,1024 @@ using namespace ::Microsoft::Console;
+            {
+                SetParent(coreHandle, _window.get());
+            }
+        }
+    }
 
-                // On Windows 11, we DONT want to refrigerate the window. There,
-                // we can just close it like normal. Break out of the loop, so
-                // we don't try to put this window in the fridge.
-                if (Utils::IsWindows11())
-                {
-                    decrementWindowCount.reset();
-                    break;
-                }
-                else
-                {
-                    window->Refrigerate();
-                    decrementWindowCount.reset();
+    // Main message loop. It pumps all windows.
+    bool loggedInteraction = false;
+    MSG msg{};
+    while (GetMessageW(&msg, nullptr, 0, 0))
+    {
+        if (!loggedInteraction && (msg.message == WM_KEYDOWN || msg.message == WM_SYSKEYDOWN))
+        {
+            TraceLoggingWrite(
+                g_hWindowsTerminalProvider,
+                "SessionBecameInteractive",
+                TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+                TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+            loggedInteraction = true;
+        }
 
-                    if (auto self{ weakThis.lock() })
-                    {
-                        auto fridge{ self->_oldThreads.lock() };
-                        fridge->push_back(window);
-                    }
-                }
+        // GH#638 (Pressing F7 brings up both the history AND a caret browsing message)
+        // The Xaml input stack doesn't allow an application to suppress the "caret browsing"
+        // dialog experience triggered when you press F7. Official recommendation from the Xaml
+        // team is to catch F7 before we hand it off.
+        // AppLogic contains an ad-hoc implementation of event bubbling for a runtime classes
+        // implementing a custom IF7Listener interface.
+        // If the recipient of IF7Listener::OnF7Pressed suggests that the F7 press has, in fact,
+        // been handled we can discard the message before we even translate it.
+        if ((msg.message == WM_KEYDOWN || msg.message == WM_SYSKEYDOWN) && msg.wParam == VK_F7)
+        {
+            if (const auto w = _mostRecentWindow(); w && w->OnDirectKeyEvent(VK_F7, LOBYTE(HIWORD(msg.lParam)), true))
+            {
+                // The application consumed the F7. Don't let Xaml get it.
+                continue;
             }
+        }
 
-            // Now that we no longer care about this thread's window, let it
-            // release it's app host and flush the rest of the XAML queue.
-            window->RundownForExit();
+        // GH#6421 - System XAML will never send an Alt KeyUp event. So, similar
+        // to how we'll steal the F7 KeyDown above, we'll steal the Alt KeyUp
+        // here, and plumb it through.
+        if ((msg.message == WM_KEYUP || msg.message == WM_SYSKEYUP) && msg.wParam == VK_MENU)

```

Comment:
this condition and the one below have the same body, roughly. can they be consolidated? i know it's not net new Leonard code